src/AutomationTests/AutomationIntegrationTests.cs

```diff
@@ -77,50 +78,119 @@ public void Cleanup()
         {
-            WindowScanOutput results = Scan_Integration_Core(WildlifeManagerAppPath, WildlifeManagerKnownErrorCount);
+            WindowScanOutput results = Scan_Integration_Core(sync, WildlifeManagerAppPath, WildlifeManagerKnownErrorCount);
             EnsureGeneratedFileIsReadableByOldVersionsOfAxeWindows(results, TestProcess.Id);
         }
 
-        // [TestMethod]
+        // [DataTestMethod]
+        //[DataRow(true)]
+        //[DataRow(false)]
         [Timeout(30000)]
-        public void Scan_Integration_Win32ControlSampler()
+        public void Scan_Integration_Win32ControlSampler(bool sync)
         {
-            Scan_Integration_Core(Win32ControlSamplerAppPath, Win32ControlSamplerKnownErrorCount);
+            Scan_Integration_Core(sync, Win32ControlSamplerAppPath, Win32ControlSamplerKnownErrorCount);
         }
 
-        [TestMethod]
+        [DataTestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
         [Timeout(30000)]
-        public void Scan_Integration_WindowsFormsControlSampler()
+        public void Scan_Integration_WindowsFormsControlSampler(bool sync)
         {
-            Scan_Integration_Core(WindowsFormsControlSamplerAppPath, WindowsFormsControlSamplerKnownErrorCount);
+            Scan_Integration_Core(sync, WindowsFormsControlSamplerAppPath, WindowsFormsControlSamplerKnownErrorCount);
         }
 
-        [TestMethod]
+        [DataTestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
         [Timeout(30000)]
-        public void Scan_Integration_WindowsFormsMultiWindowSample()
+        public void Scan_Integration_WindowsFormsMultiWindowSample(bool sync)
         {
-            Scan_Integration_Core(WindowsFormsMultiWindowSamplerAppPath, WindowsFormsMultiWindowSamplerAppAllErrorCount, true, 2);
+            Scan_Integration_Core(sync, WindowsFormsMultiWindowSamplerAppPath, WindowsFormsMultiWindowSamplerAppAllErrorCount, true, 2);
         }
 
-        [TestMethod]
+        [DataTestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        [Timeout(30000)]
+        public void Scan_Integration_WindowsFormsMultiWindowSample_SingleWindow(bool sync)
+        {
+            Scan_Integration_Core(sync, WindowsFormsMultiWindowSamplerAppPath, WindowsFormsMultiWindowSamplerSingleWindowAllErrorCount);
+        }
+
+        [DataTestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
         [Timeout(30000)]
-        public void Scan_Integration_WindowsFormsMultiWindowSample_SingleWindow()
+        public void Scan_Integration_WpfControlSampler(bool sync)
         {
-            Scan_Integration_Core(WindowsFormsMultiWindowSamplerAppPath, WindowsFormsMultiWindowSamplerSingleWindowAllErrorCount);
+            Scan_Integration_Core(sync, WpfControlSamplerAppPath, WpfControlSamplerKnownErrorCount);
         }
 
         [TestMethod]
         [Timeout(30000)]
-        public void Scan_Integration_WpfControlSampler()
+        public void Scan_Throws_When_Cancelled()
```

Comment:
Since `Scan` can't cancel, should we rename this to `ScanAsync_Throws_When_Cancelled` ?